assets/src/edit-story/elements/image/output.js

```diff
@@ -42,6 +44,15 @@ function ImageOutput({ element, box }) {
   const srcSet = calculateSrcSet(element.resource);
   if (srcSet) {
     props.srcSet = srcSet;
+
+    // If `srcset` exists but `sizes` doesn't, amp-img will generate a sizes attribute
+    // with best-guess values that can result in poor image selection.
+    const imageWidthPercent = element.width / PAGE_WIDTH;
+    const mobileWidth = Math.round(imageWidthPercent * 100) + 'vw';
+    // Width of a story page in desktop mode is 45vh.
+    const desktopWidth = Math.round(imageWidthPercent * 45) + 'vh';
+    // 1024px is the minimum width for STAMP desktop mode.
+    props.sizes = `(min-width: 1024px) ${desktopWidth}, ${mobileWidth}`;
```

Comment:
This is very different from how WordPress uses the size attributes. Take this example. 

```
<img loading="lazy" width="1024" height="640" src="http://localhost:8899/wp-content/uploads/2021/06/download-1024x640.jpeg" alt="" class="wp-image-1021" srcset="http://localhost:8899/wp-content/uploads/2021/06/download-1024x640.jpeg 1024w, http://localhost:8899/wp-content/uploads/2021/06/download-300x188.jpeg 300w, http://localhost:8899/wp-content/uploads/2021/06/download-768x480.jpeg 768w, http://localhost:8899/wp-content/uploads/2021/06/download-1536x960.jpeg 1536w, http://localhost:8899/wp-content/uploads/2021/06/download-1200x750.jpeg 1200w, http://localhost:8899/wp-content/uploads/2021/06/download-150x94.jpeg 150w, http://localhost:8899/wp-content/uploads/2021/06/download.jpeg 1920w" sizes="(max-width: 1024px) 100vw, 1024px">
```

It uses a max-width that is the same size as the source image. 

Can you explain the reasoning being this size.

Reply:
Sure, `min-width: 1024px` comes from the STAMP runtime that toggles the desktop viewing mode: https://github.com/ampproject/amphtml/blob/f8f7c7dbec61cec4602e47b00dc883f9bc68ded1/extensions/amp-story/1.0/amp-story.js#L135

That means at screen widths >= 1024px, the browser should use `desktopWidth` as the width of the image for `srcset` selection.